ocaml/xapi/helpers.ml

```diff
@@ -1849,55 +1855,80 @@ end = struct
       in
       let conf_contents =
         String.concat "\n"
-          [
-            "; autogenerated by xapi"
-          ; Printf.sprintf "fips = %s" fips
-          ; "pid = /var/run/xapissl.pid"
-          ; "socket = r:TCP_NODELAY=1"
-          ; "socket = a:TCP_NODELAY=1"
-          ; "socket = l:TCP_NODELAY=1"
-          ; "socket = r:SO_KEEPALIVE=1"
-          ; "socket = a:SO_KEEPALIVE=1"
-          ; ( match Sys.getenv_opt "STUNNEL_IDLE_TIMEOUT" with
+          ([
+             "; autogenerated by xapi"
+           ; Printf.sprintf "fips = %s" fips
+           ; "pid = /var/run/xapissl.pid"
+           ; "socket = r:TCP_NODELAY=1"
+           ; "socket = a:TCP_NODELAY=1"
+           ; "socket = l:TCP_NODELAY=1"
+           ; "socket = r:SO_KEEPALIVE=1"
+           ; "socket = a:SO_KEEPALIVE=1"
+           ; "TIMEOUTclose = 1"
+           ; ( match Sys.getenv_opt "STUNNEL_IDLE_TIMEOUT" with
+             | None ->
+                 "; no idle timeout"
+             | Some x ->
+                 Printf.sprintf "TIMEOUTidle = %s" x
+             )
+           ; Stunnel.debug_conf_of_env ()
+           ; "protocol = proxy" (* tells stunnel to include inet address info *)
+           ; ""
+           ; "[xapi]"
+           ; Printf.sprintf "accept = %s" accept
+           ; Printf.sprintf "cert = %s" cert
+           ; Printf.sprintf "ciphers = %s" Xcp_const.good_ciphersuites
+           ; "curve = secp384r1"
+           ; "options = CIPHER_SERVER_PREFERENCE"
+           ; "sslVersion = TLSv1.2"
+           ]
+          @ ( match client_auth_name with
```

Comment:
I think we should use `List.concat` on the other level and avoid `@` here.

Reply:
Then `@` becomes `;`, but is that clearer/better?

Reply:
With `@` you have to be careful about precedence if you are combine it with `if-then-else`.